src/grt/src/GlobalRouter.cpp

```diff
@@ -2632,18 +2609,18 @@ bool GlobalRouter::isClkTerm(odb::dbITerm* iterm, sta::dbNetwork* network)
   return lib_port->isRegClk();
 }
 
-bool GlobalRouter::clockHasLeafITerm(odb::dbNet* db_net)
+bool GlobalRouter::isNonLeafClock(odb::dbNet* db_net)
 {
   sta::dbNetwork* network = sta_->getDbNetwork();
   if (db_net->getSigType() == odb::dbSigType::CLOCK) {
     for (odb::dbITerm* iterm : db_net->getITerms()) {
       if (isClkTerm(iterm, network)) {
-        return true;
+        return false;
       }
     }
   }
 
-  return false;
+  return true;
```

Comment:
If something is not a clock this should return false (it can't be a nonLeafClock if it isn't a clock at all).  It would also simplify to not required the &&...

 bool is_non_leaf_clock = isNonLeafClock(net->getDbNet()) && is_clock;

Reply:
This is fine but I would have reversed the initial statement:

 if (db_net->getSigType() != odb::dbSigType::CLOCK) {
  return false;
}

it is less nesting and simpler to read.